frontend/public/components/start-guide.tsx

```diff
@@ -11,10 +11,24 @@ import { Disabled, HintBlock, ExternalLink, openshiftHelpBase, LinkifyExternal }
 import { connectToFlags } from '../reducers/connectToFlags';
 import { ProjectModel } from '../models';
 import { createProjectModal } from './modals/create-namespace-modal';
+import { useActivePerspective } from '@console/dynamic-plugin-sdk';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useActiveNamespace } from '@console/shared';
 
 export const OpenShiftGettingStarted = connect(createProjectMessageStateToProps)(
   ({ canCreateProject = true, createProjectMessage }: OpenShiftGettingStartedProps) => {
     const { t } = useTranslation();
+    const [, setActiveNamespace] = useActiveNamespace();
+    const [perspective] = useActivePerspective();
+
+    const handleSubmit = React.useMemo(() => {
+      if (perspective === 'dev') {
+        return (project: K8sResourceKind) => {
+          setActiveNamespace(project.metadata?.name);
+        };
+      }
+    }, [perspective, setActiveNamespace]);
```

Comment:
You can use `useCallback` which is more common for a callback ;)

And then let us use the "old behavior" only on the admin perspective, and set the namespace on all other perspectives.

```suggestion
    const handleSubmit = React.useCallback((project: K8sResourceKind) => {
      if (perspective !== 'admin') {
        setActiveNamespace(project.metadata?.name);
      }
    }, [perspective, setActiveNamespace]);
```